txp/Pos/Txp/Toil/Trans.hs

```diff
@@ -42,8 +45,16 @@ instance MonadUtxoRead m => MonadUtxoRead (ToilT __ m) where
     utxoGet id = ether $ MM.lookupM utxoGet id =<< use tmUtxo
 
 instance MonadUtxoRead m => MonadUtxo (ToilT __ m) where
-    utxoPut id aux = ether $ tmUtxo %= MM.insert id aux
-    utxoDel id = ether $ tmUtxo %= MM.delete id
+    utxoPut id aux = ether $ do
+        utxo <- use tmUtxo
+        when (id `HM.member` MM.insertionsMap utxo) $
+            error ("utxoPut@ToilT: "+|id|+" is already in utxo")
+        tmUtxo %= MM.insert id aux
+    utxoDel id = ether $ do
+        utxo <- use tmUtxo
+        when (id `HS.member` MM.deletionsSet utxo) $
+            error ("utxoDel@ToilT: "+|id|+" is already deleted from utxo")
+        tmUtxo %= MM.delete id
```

Comment:
Isn't it better to use `utxoGet` to detect whether we have input in underlying storage?
In this case you can define this logic in general way and leave old functions in class.
E. g. have `utxoPutUnchecked` and `utxoDelUnchecked` inside the class and `utxoPut` and `utxoDel` outside.

Reply:
:+1: